### PR TITLE
feat/mock-server/setup

### PR DESCRIPTION
**Pull request by sandrodesouza**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 Common test suit for all backend SDKs
+
+# TODO
+
+-   [ ] check for FIXME on api-mock folder
+-   [ ] not sure why python didn't worked with my default docker startup, meanwhile node did
+-   [ ] how to handle mock-server shutdown when tests fail
+-   [ ] mocha .only isn't working when not specifying the test file
+-   [ ] move api-mock-server to their own project
+-   [ ] migrate accountlinking, mfa, accountlinking-with-session tests
```
Comment (rishabhpoddar) [inline README.md:12]: Make a PR from base -> main, and then add this list in there, instead of the readme.

File: api-mock/api-mock-server.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+import json
```
Comment (rishabhpoddar) [inline api-mock/api-mock-server.py:1]: mock server should go in python sdk

File: api-mock/api-mock-server.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import debug from "debug";
```
Comment (rishabhpoddar) [inline api-mock/api-mock-server.ts:1]: mock server should go in node sdk

File: hooks/pre-commit.sh [hunk 4]
```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
```
Comment (rishabhpoddar) [inline hooks/pre-commit.sh:1]: probably needs changing

File: start-mock-server.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```
Comment (rishabhpoddar) [inline start-mock-server.sh:1]: i prefer if we do not run stuff in the background. Easier to make sure it's going to work in all OS. The user can just start the server manually, and the press cmd+c to stop it.

File: api-mock/index.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
+let EmailPassword = require("supertokens-node/recipe/emailpassword");
```
Comment (rishabhpoddar) [inline api-mock/index.js:1]: can we make this TS?

File: version.js [hunk 7]
```diff
@@ -0,0 +1 @@
+exports.version = "17.1.0";
```
Comment (rishabhpoddar) [inline version.js:1]: what is this file for?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api-mock/api-mock-server.py (hunk 2 above)
Comment (sandrodesouza) [reply api-mock/api-mock-server.py:1]: yes, but I'm waiting for the final version to move it

File: api-mock/api-mock-server.ts (hunk 3 above)
Comment (sandrodesouza) [reply api-mock/api-mock-server.ts:1]: same, waiting for the final version

File: api-mock/index.js (hunk 6 above)
Comment (sandrodesouza) [reply api-mock/index.js:1]: yes but we will need to run a build first or figure it out on how mocha runs ts

File: start-mock-server.sh (hunk 5 above)
Comment (sandrodesouza) [reply start-mock-server.sh:1]: ok

File: README.md (hunk 1 above)
Comment (sandrodesouza) [reply README.md:12]: ok

File: version.js (hunk 7 above)
Comment (sandrodesouza) [reply version.js:1]: this file was need by pre-commit hook, I will probably remove it when implement versioning

**Review by porcellus — APPROVED** [review]
I think by and large this is good (with some issues it has addressed by the other PR). You can merge this into the base branch whenever you prefer or merge it with the other PR (that may be easier to review).

**Pull request merged**